code/datums/gameticker.dm

```diff
@@ -766,7 +766,19 @@ var/global/current_state = GAME_STATE_WORLD_INIT
 
 	//logTheThing("debug", null, null, "Zamujasa: [world.timeofday] finished spacebux updates")
 
-
+	var/list/playtimes = list() //associative list with the format list("ckeys\[[player_ckey]]" = playtime_in_seconds)
+	for(var/datum/player/P in by_type[/datum/player])
+		if (!P || !P.ckey)
```

Comment:
```suggestion
		if (!P.ckey)
```
You are iterating with a typed variable so P is always of type `/datum/player` and as such can't be null.